src/on.ts

```diff
@@ -0,0 +1,93 @@
+import { Handle, EventObject } from './interfaces';
+import { createHandle, createCompositeHandle } from './lang';
+import Evented from './Evented';
+
+// Only used for Evented and EventEmitter, as EventTarget uses EventListener
+type EventCallback = (event: {}) => void;
+
+interface ExtensionEvent {
+	(target: any, listener: EventCallback): Handle;
+}
+
+interface EventTarget {
+	addEventListener(event: string, listener: EventListener, capture?: boolean): Handle;
+	removeEventListener(event: string, listener: EventListener, capture?: boolean): void;
+}
+
+interface EventEmitter {
+    on(event: string, listener: EventCallback): EventEmitter;
+	removeListener(event: string, listener: EventCallback): EventEmitter;
+}
+
+export default function on(target: EventTarget, type: string, listener: EventListener, capture?: boolean): Handle;
+export default function on(target: EventTarget, type: ExtensionEvent, listener: EventListener, capture?: boolean): Handle;
+export default function on(target: EventTarget, type: [string|ExtensionEvent], listener: EventListener, capture?: boolean): Handle;
+
+export default function on(target: EventEmitter, type: string, listener: EventCallback): Handle;
+export default function on(target: EventEmitter, type: ExtensionEvent, listener: EventCallback): Handle;
+export default function on(target: EventEmitter, type: [string|ExtensionEvent], listener: EventCallback): Handle;
+
+export default function on(target: Evented, type: string, listener: EventCallback): Handle;
+export default function on(target: Evented, type: ExtensionEvent, listener: EventCallback): Handle;
+export default function on(target: Evented, type: [string|ExtensionEvent], listener: EventCallback): Handle;
+
+export default function on(target: any, type: any, listener: EventListener, capture?: boolean): Handle {
+	if (type.call) {
+		return type.call(this, target, listener, capture);
+	}
+
+	if (!!type.shift) {
+		var handles: Handle[] = type.map(function (type: string): Handle {
+			return on(target, type, listener, capture);
+		});
+
+		return createCompositeHandle.apply(null, handles);
+	}
+
+	if (target.addEventListener && target.removeEventListener) {
+		target.addEventListener(type, listener, false);
```

Comment:
Two things:

1. The third argument needs to be `capture`.
2. If I pass the same listener into this function twice, `addEventListener` will not register it a second time. The listener should probably be wrapped with a function to make it unique.